### PR TITLE
GIDS: Seed Accreditation file type data#15765

### PR DESCRIPTION
**Pull request by ksantiagoBAH**
## Description
As a developer, I need to seed accreditation file type data so that there is test data available in the local environment for developer and QA team members to use in future stories.

https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/15765

## Testing done
local testing

## Screenshots
Screen Shot 2020-11-12 at 1 52 10 PM

## Acceptance criteria
- [x] The local development environment has Accreditation file type seed data.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: lib/seed_utils.rb [hunk 1]
```diff
@@ -3,6 +3,41 @@
...
+    group_options[:types].each do |type|
+      sheets << {
+        klass: type,
+        skip_lines: 0
+      }
+    end
+
+    file_options = { sheets: sheets }
+    csv_name = "#{group}.xlsx"
+    csv_path = 'sample_csvs'
```
Comment (zurbergram) [inline lib/seed_utils.rb:18]: can you pull `csv_path` out into a class variable?

File: lib/seed_utils.rb [hunk 2]
```diff
@@ -33,6 +68,11 @@ def seed_table_with_upload(klass, user, options = {})
     puts 'Done!'
   end
 
+  def load_table(klass, _user, _file_options, csv_name)
```
Comment (zurbergram) [inline lib/seed_utils.rb:71]: what is this method?

File: lib/seed_utils.rb [hunk 3]
```diff
@@ -3,6 +3,41 @@
...
+      }
+    end
+
+    file_options = { sheets: sheets }
+    csv_name = "#{group}.xlsx"
+    csv_path = 'sample_csvs'
+
+    load_table(Group, user, file_options.reverse_merge(options), csv_name)
+
+    uf = ActionDispatch::Http::UploadedFile.new(
```
Comment (zurbergram) [inline lib/seed_utils.rb:22]: this should be pulled out into its own method
`def upload_file(name, content_type)`

File: lib/seed_utils.rb [hunk 4]
```diff
@@ -12,7 +47,7 @@ def seed_table_with_upload(klass, user, options = {})
     csv_name = "#{csv_type.underscore}.csv"
     csv_path = 'sample_csvs'
 
-    puts "Loading #{klass.name} from #{csv_path}/#{csv_name} ... "
+    load_table(klass, user, file_options, csv_name)
 
     uf = ActionDispatch::Http::UploadedFile.new(
```
Comment (zurbergram) [inline lib/seed_utils.rb:52]: new method for creating the upload file object
`def upload_file(name, content_type)`

**Review by zurbergram — CHANGES_REQUESTED** [review]
The code in `seed_tables_with_group` and `seed_table_with_upload` is very similar, consider moving similar code into `load_table`

File: lib/seed_utils.rb [hunk 5]
```diff
@@ -33,6 +68,11 @@ def seed_table_with_upload(klass, user, options = {})
     puts 'Done!'
   end
 
+  def load_table(klass, _user, _file_options, csv_name)
+    csv_path = 'sample_csvs'
```
Comment (shawkey-daniel) [inline lib/seed_utils.rb:72]: This looks like a dup of line 13.

File: lib/seed_utils.rb (hunk 2 above)
Comment (shawkey-daniel) [inline lib/seed_utils.rb:71]: Is _user and _file_options used here?

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]

File: lib/seed_utils.rb (hunk 2 above)
Comment (ksantiagomorgan6) [reply lib/seed_utils.rb:71]: I will be now, I'm breaking part of the code out into a new function.

**Review by zurbergram — APPROVED** [review]

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]
👍

File: db/seeds/02_data.rb [hunk 6]
```diff
@@ -34,7 +34,8 @@
   puts 'Deleting old versions'
   Version.delete_all
 
-  puts 'Loading CSVs. Why not do some calf raises while you wait? ... '
+  puts 'Loading CSVs. Why not do some calf raises while you wait? ...' 
+  SeedUtils.seed_tables_with_group('Accreditation', user)
```
Comment (kfrz) [inline db/seeds/02_data.rb:38]: Why is this instance of the SeedUtils method called using a string instead of class-name?

**Review by kfrz — APPROVED** [review]
This looks good, one minor question. Also looks like codeclimate has an issue

File: db/seeds/02_data.rb (hunk 6 above)
Comment (ksantiagomorgan6) [reply db/seeds/02_data.rb:38]: Accreditation isn't a class in this case so the string `'Accreditation'` is used to upload the excel file.

**Pull request merged**